src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -58,7 +58,7 @@ internal sealed partial class ImmutabilityContext {
 		public bool IsImmutable(
 			ITypeSymbol type,
 			ImmutableTypeKind kind,
-			Location location,
+			Func<Location> locationGetter,

```

Comment:
nit: getLocation to match `GetLocation` + the active voice for function names (IsImmutable, CheckDefinition, etc.)

Reply:
It's not a method, but I try (but sometimes probably fail) to name my funcs as if they were: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods